packages/sdk/src/multicall.ts

```diff
@@ -18,29 +18,29 @@ export type EncodedRequest = {
   callData: string;
 };
 
-interface State {
+export interface State {
   requests: Request[];
-  multicallClient: multicall.Instance;
+  multicallClient: multicall.Instance | multicall2.Instance;
```

Comment:
I think this might work, make state have a generic:
```ts
interface State <MulticallType>{
   multicallClient: MulticallType
}
```

Then in the class

```ts
class Multicall <MulticallType> implements State {
  public requests: Request[];
  public multicallClient: MulticallType;
```

Then in factory:
```ts
export default class Factory extends Multicall<multicall.Instance> {
  constructor(address: string, provider: SignerOrProvider) {
    const multicallClient = multicall.connect(address, provider);
    super({ multicallClient, requests: [] });
  }
}
```

I dont know if this actually compiles, but it might work, wdyt?

Reply:
I agree that multicall2.Instance should be pulled out of Multicall. I'll work smth out base on your generics suggestion